consensus/parlia/snapshot.go

```diff
@@ -219,7 +276,14 @@ func (s *Snapshot) apply(headers []*types.Header, chain consensus.ChainHeaderRea
 				}
 			}
 			snap.Validators = newVals
+			if chainConfig.IsBoneh(header.Number) {
```

Comment:
Why dont you move it to when you init the `ValidatorInfo`, also have `IsBoneh` condition.